### PR TITLE
Visibility per motion property

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

1. MotionProperty exposes its visibility as a flow. Snap, animateTo, lerpTo automatically trigger visibility updates
2. Motion property allows to override the source of data from which it resolves its visibility 
3. AnimatingState considers geometry animations

## Check list

- [ ] I have updated `CHANGELOG.md` if required.
- [ ] I have updated documentation if required.

### CODE REVIEW

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/MotionProperty.kt [hunk 1]
```diff
@@ -121,10 +152,10 @@ abstract class MotionProperty<T, V : AnimationVector>(
         return velocity
     }
 
-    suspend fun animateTo(
+    open suspend fun animateTo(
```
Comment (KovalevAndrey) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/MotionProperty.kt]: remove

File: appyx-interactions/android/src/androidTest/kotlin/com/bumble/appyx/interactions/ComposeContentTestRuleExt.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.bumble.appyx.interactions
+
+import androidx.compose.ui.test.junit4.ComposeContentTestRule
+import com.bumble.appyx.interactions.core.InteractionModel
+
+fun InteractionModel<*, *>.ensureAnimationStarted(rule: ComposeContentTestRule) {
```
Comment (KovalevAndrey) [inline appyx-interactions/android/src/androidTest/kotlin/com/bumble/appyx/interactions/ComposeContentTestRuleExt.kt]:
```suggestion
fun InteractionModel<*, *>.waitUntilAnimationStarted(rule: ComposeContentTestRule) {
```

File: appyx-interactions/android/src/androidTest/kotlin/com/bumble/appyx/interactions/transitionmodel/spotlight/SpotlightTest.kt [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        @Parameterized.Parameters
+        fun data() = arrayOf(
+            TestParam(operationMode = Operation.Mode.IMMEDIATE),
+            TestParam(operationMode = Operation.Mode.KEYFRAME),
+            TestParam(operationMode = Operation.Mode.GEOMETRY)
+        )
+    }
+
+    @Test
+    fun spotlight_clip_toBounds() {
```
Comment (KovalevAndrey) [inline appyx-interactions/android/src/androidTest/kotlin/com/bumble/appyx/interactions/transitionmodel/spotlight/SpotlightTest.kt]:
```suggestion
    fun spotlight_calculates_visibile_elements_correctly_when_clipToBounds_is_false() {
```

File: appyx-interactions/android/src/androidTest/kotlin/com/bumble/appyx/interactions/transitionmodel/spotlight/SpotlightTest.kt [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    val composeTestRule = createComposeRule()
+
+    private lateinit var spotlight: Spotlight<NavTarget>
+
+    companion object {
+
+        data class TestParam(
+            val operationMode: Operation.Mode = Operation.Mode.IMMEDIATE,
+            val clipToBounds: Boolean = false,
+            val containerWidthFraction: Float = 1.0f,
```
Comment (KovalevAndrey) [inline appyx-interactions/android/src/androidTest/kotlin/com/bumble/appyx/interactions/transitionmodel/spotlight/SpotlightTest.kt]: remove

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/transitionmodel/spotlight/interpolator/SpotlightSlider.kt [hunk 5]
```diff
@@ -92,93 +84,68 @@ class SpotlightSlider<InteractionTarget : Any>(
...
-        offset = OffsetP(DpOffset.Zero).also {
-            it.displacement = derivedStateOf {
-                DpOffset((scroll.value * width.value).dp, 0.dp)
-            }
-        },
-        screenWidth = screenWidth,
-        containerWidth = width,
-        clipToBounds = clipToBounds,
-        transitionBounds = transitionBounds
+    var position = 0
```
Comment (KovalevAndrey) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/transitionmodel/spotlight/interpolator/SpotlightSlider.kt]: delete

**Review by zsoltk — APPROVED** [review]

**Pull request merged**